tests/receivers/oracledb/bundled_test.go

```diff
@@ -33,29 +35,85 @@ func TestOracledbDockerObserver(t *testing.T) {
 		dockerSocket.Stop()
 	})
 
-	testutils.AssertAllMetricsReceived(t, "bundled.yaml", "otlp_exporter.yaml",
-		nil, []testutils.CollectorBuilder{
-			func(c testutils.Collector) testutils.Collector {
-				cc := c.(*testutils.CollectorContainer)
-				cc.Container = cc.Container.WillWaitForLogs("Discovering for next")
-				return cc
-			},
-			func(collector testutils.Collector) testutils.Collector {
-				return collector.WithEnv(map[string]string{
-					"SPLUNK_DISCOVERY_DURATION": "20s",
-					// confirm that debug logging doesn't affect runtime
-					"SPLUNK_DISCOVERY_LOG_LEVEL": "debug",
-					"ORACLE_PASSWORD":            "password",
-				}).WithArgs(
-					"--discovery",
-					"--set", "splunk.discovery.receivers.oracledb.config.username=otel",
-					"--set", "splunk.discovery.receivers.oracledb.config.password='${ORACLE_PASSWORD}'",
-					"--set", "splunk.discovery.receivers.oracledb.config.service=XE",
-					"--set", `splunk.discovery.extensions.k8s_observer.enabled=false`,
-					"--set", `splunk.discovery.extensions.host_observer.enabled=false`,
-					"--set", fmt.Sprintf("splunk.discovery.extensions.docker_observer.config.endpoint=tcp://%s", dockerSocket.ContainerEndpoint),
-				)
-			},
-		},
-	)
+	tc := testutils.NewTestcase(t)
+	defer tc.PrintLogsOnFailure()
+	defer tc.ShutdownOTLPReceiverSink()
+
+	_, shutdown := tc.SplunkOtelCollectorContainer("otlp_exporter.yaml", func(c testutils.Collector) testutils.Collector {
+		cc := c.(*testutils.CollectorContainer)
+		cc.Container = cc.Container.WillWaitForLogs("Discovering for next")
+		return cc
+	},
+		func(collector testutils.Collector) testutils.Collector {
+			return collector.WithEnv(map[string]string{
+				"SPLUNK_DISCOVERY_DURATION": "20s",
+				// confirm that debug logging doesn't affect runtime
+				"SPLUNK_DISCOVERY_LOG_LEVEL": "debug",
+				"ORACLE_PASSWORD":            "password",
+			}).WithArgs(
+				"--discovery",
+				"--set", "splunk.discovery.receivers.oracledb.config.username=otel",
+				"--set", "splunk.discovery.receivers.oracledb.config.password='${ORACLE_PASSWORD}'",
+				"--set", "splunk.discovery.receivers.oracledb.config.service=XE",
+				"--set", `splunk.discovery.extensions.k8s_observer.enabled=false`,
+				"--set", `splunk.discovery.extensions.host_observer.enabled=false`,
+				"--set", fmt.Sprintf("splunk.discovery.extensions.docker_observer.config.endpoint=tcp://%s", dockerSocket.ContainerEndpoint),
+			)
+		})
+	defer shutdown()
+
+	metricNames := []string{
```

Comment:
Is there a reason for this format vs. a golden metrics file?

Reply:
I can't run the test locally to capture the oracledb golden file because the oracledb container refuses to work on an ARM architecture in a weird way. I ended up copying what was in the resource_metrics file. The other test checks that all metrics are reported, so we still have coverage.